code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -713,9 +714,20 @@ butcher
 	var/run_range = 3
 	var/list/dummy_params = list("icon-x" = 16, "icon-y" = 16)
 
+/datum/aiTask/timed/targeted/flockdrone_shoot/proc/precondition()
+	//if we know there are enemies, and gun is charged and ready
+	var/mob/living/critter/flock/drone/F = holder.owner
+	if(length(F.flock?.enemies))
+		var/datum/handHolder/HH = F.hands[3]
+		var/datum/limb/gun/stunner = HH?.limb
+		if(istype(stunner) && !stunner.is_on_cooldown())
+			return TRUE
 
 /datum/aiTask/timed/targeted/flockdrone_shoot/evaluate()
-	return weight * score_target(get_best_target(get_targets()))
+	if(src.precondition())
```

Comment:
Surely this check is already done by the AI holder?

Reply:
it is not. Precondition is defined for sequential aitasks, but not for timed for some bizarre reason.

Reply:
![image](https://user-images.githubusercontent.com/20713227/168443232-fd894de3-f522-4fd2-88c9-222b840ec8ea.png)